src/app/(landing-page)/blog/[posts]/page.tsx

```diff
@@ -0,0 +1,43 @@
+"use client";
+
+import { Container } from "@mui/material";
+import BlogPost from "components/BlogPage/BlogPost";
+import { IBlog } from "components/BlogPage/types";
+import { LoadingScreen } from "components/global-components";
+import { doc, getDoc } from "firebase/firestore";
+import { db } from "lib/firebase/config";
+import { notFound } from "next/navigation";
+import { FC, useEffect, useState } from "react";
+
+interface pageProps {
+  params: { posts: string };
+}
+
+const Post: FC<pageProps> = ({ params }) => {
+  const [loadedDoc, setLoadedDoc] = useState<IBlog | null>(null);
+
+  useEffect(() => {
+    const getBlog = async () => {
+      const docRef = doc(db, "blogs", params.posts);
+      const docSnap = await getDoc(docRef);
+      if (docSnap.exists()) {
+        setLoadedDoc(docSnap.data() as IBlog);
+      } else {
+        notFound();
+      }
+    };
+
+    getBlog();
+  }, [params.posts]);
+
+  if (loadedDoc) {
```

Comment:
could you change this to if (!loadedDoc) return Loading, just to reduce nested code